### ISSUE TITLE
Notification Colors - Icon colors against notification background colors don't meet accessibility

### ISSUE THREAD

**Issue by KennyAtHPE** [grommet/hpe-design-system#2446]
Related to #2425 

![image](https://user-images.githubusercontent.com/86075038/157741051-79e2bc4c-73bc-4286-9b14-7d315253ec2e.png)
Text color - default text
Background colors - Validation Warning, Validation Critical, Background Contrast, Validation OK
Status icon colors - Status Warning, Status Critical, Text Strong, Status OK

Concerns relating to the current dark mode validation and status colors:
- Default text is hard to see with the current validation colors
- Status colors blend in with validation colors
- Whether or not they do a good enough job capturing user's attention

This is affecting use cases such as global banners and form validation

## Updated Success Criteria

- Update Status icon and foreground colors, so that when used against validation and default backgrounds in both light and dark themes, shall achieve a minimum contrast ratio of 3:1 ([Accessibility rules](https://www.w3.org/WAI/WCAG21/Techniques/general/G207)).

**Comment by vavalos5** [comment]
@KennyAtHPE Were there colors you had explored as alternative colors for this ticket?

**Comment by KennyAtHPE** [comment]
@vavalos5 there's a very small amount of exploration [here](https://www.figma.com/file/FGvOd2htjrChVVS2oz73ds/HPE-Notifications?node-id=3597%3A16812). The colors used here are all already within our current color palette. Ultimately, we moved away from this exploration since we didn't want to use different existing colors but instead are advocating that the highlighted colors within this ticket be changed instead.

**Comment by vavalos5** [comment]
Thanks Kenny. Once Chris returns, i'll chat w/ him on this matter. If it's a priority @SOjaHPE

**Comment by vavalos5** [comment]
We should do more research and data 
"not to be so rainbow-ish" - we can bring colors up and down
are we making the styles aligned and that it should react like the coded component

**Comment by vavalos5** [comment]
An idea I came across which we can possibly consider
![Screen Shot 2022-04-15 at 2 13 41 PM](https://user-images.githubusercontent.com/62117454/164106979-48ee4ac5-f638-48f5-83d7-fb3c4c379435.png)

**Comment by vavalos5** [comment]
Sherry Mead reached out today to see if we have a solution. Don't know if we want to bump this up

**Comment by SmWorkGitHub** [comment]
Just wanted to repeat the issue description I gave Vicky in Figma. It applies to light and dark mode validation colors: 

> I'm going to be asking the owners of a UI component library to change the validation behavior of their form components and they will push back on semi transparent form field backgrounds for the reason Matt gave. 
> 
> They can't control what teams do with their components so the components have to be readable and pass accessibility even if teams place them on colored or patterned backgrounds.

**Comment by ericsoderberghp** [comment]
These colors were updated as part of https://github.com/grommet/grommet-theme-hpe/pull/280. We should re-check if the issue still exists.

**Comment by taysea** [comment]
@KennyAtHPE @oliverHPE before I close this, can you assess the current values and confirm if we've resolved this issue?

**Comment by oliverHPE** [comment]
Thanks @taysea Have found several colours for each validation type and need to confirm which are the latest.

**Comment by KennyAtHPE** [comment]
@taysea @oliverHPE I've reviewed all the different states within our notifications and here's what I've found:
- Current text color is now accessible on all the different state backgrounds within the dark theme
- The dark theme Icons are NOT accessible in all four states

WCAG states "if the icons are required to understand the content, then the icons need to have a contrast ratio of at least 3:1." [Link to docs](https://www.w3.org/WAI/WCAG21/Techniques/general/G207). I suggest we reformat this ticket with this new scope in mind.

**Comment by oliverHPE** [comment]
@taysea Thanks @KennyAtHPE Also had a look at our status colors as measured against validation backgrounds in the light theme, and these also do not meet a 3 : 1 contrast ratio.

**Comment by luketa8** [comment]
Updating:

- Added a complete [assessment](https://www.figma.com/design/6oYsSnnf9j8BJDhFwzNCWq/%232446---Notifications-accesibility?node-id=10-4301&t=egrhTq8lwQr65Zlk-4) of current icon/background pairs with contrast ratios.

- Added [alternative color explorations](https://www.figma.com/design/6oYsSnnf9j8BJDhFwzNCWq/%232446---Notifications-accesibility?node-id=21-1481&node-type=frame&t=egrhTq8lwQr65Zlk-11) that meet the 3:1 ratio

Tagging @oliverHPE and @SeamusLeonardHPE for visibility

**Comment by luketa8** [comment]
Updating Success Criteria:

- Status icon colors, when used against validation and default backgrounds in both light and dark themes, shall achieve a minimum contrast ratio of 3:1. This applies to all instances of status icons across the design system.

[ ] Provide viable options within existing design system colors thata meet

**Comment by luketa8** [comment]
- Finished [new color Tokens](https://www.figma.com/design/6oYsSnnf9j8BJDhFwzNCWq/%232446---Notifications-accesibility?node-id=160-2027), reviewed with @oliverHPE. 

- Next steps: Final signoff from @SeamusLeonardHPE. Updating the [Primitive](https://www.figma.com/design/LU0Ch8dFU6rinUaQ3zXUp5/Design-Tokens---Primitives?node-id=0-1&node-type=canvas&t=xmwymBh96tabQQcy-0) and [Semantic](https://www.figma.com/design/vH5MOEoXU5Kv7qWvBqiUC2/Design-Tokens---Semantic?node-id=0-1&node-type=canvas&t=yOSzwRCNMvU48CLA-0) Token Libraries. 

- Follow up: Ticket to make the same updates in code @jcfilben.

**Comment by jcfilben** [comment]
Can we also run the new notification colors by Chris, Joelle, and Ashley to give them visibility?

**Comment by luketa8** [comment]
Thanks Jessica. I 've sent a Slack message to @joellegregory.

**Comment by luketa8** [comment]
- Added Accesible Info (Blue) color new token -> color.blue-600
- Oliver updated Semantic and Primitive Token libraries. 

- Moving to Done and Opening follow up task to update colors in code
